File: php/features.php
<?php
declare(strict_types=1);

// TruthOrDare SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class TruthOrDareFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new TruthOrDareBaseFeature();
            case "test":
                return new TruthOrDareTestFeature();
            default:
                return new TruthOrDareBaseFeature();
        }
    }
}

File: py/truthordare_sdk/features.py
# TruthOrDare SDK feature factory

from truthordare_sdk.feature.base_feature import TruthOrDareBaseFeature
from truthordare_sdk.feature.test_feature import TruthOrDareTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TruthOrDareBaseFeature(),
        "test": lambda: TruthOrDareTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
